### PR TITLE
Find the decision for multiline iconed label

### LINKED ISSUE #863: Find the decision for multiline iconed label
Image
Image

### PR DESCRIPTION
**Pull request by LamaEats**
## PR includes

- [x] Bug Fix

## Related issues

Resolve #863 

#### Checkboxes
image
image
image

#### Radios
image
image

#### Badges
image

### CODE REVIEW

File: src/harmony/Badge/Badge.module.css [hunk 1]
```diff
@@ -1,24 +1,28 @@
...
     border-radius: var(--radius-xl);
 }
 
-.BadgeIconLeft {
-    margin-right: var(--gap-s);
-    display: inline-flex;
+.BadgeIcon {
+    display: inline-block;
     line-height: 1;
+    width: 1em;
```
Comment (awinogradov) [inline src/harmony/Badge/Badge.module.css:15]: https://github.com/taskany-inc/icons/blob/main/src/components/BaseIcon.tsx#L54

same style on BaseIcon. Why we need copy here?

File: src/harmony/Checkbox/Checkbox.module.css [hunk 2]
```diff
@@ -1,9 +1,9 @@
 .Checkbox {
     position: relative;
     display: inline-flex;
-    align-items: center;
     cursor: pointer;
     color: var(--text-primary);
+    font-size: var(--font-size-m);
```
Comment (awinogradov) [inline src/harmony/Checkbox/Checkbox.module.css:6]: why not inherited?

File: src/harmony/Checkbox/Checkbox.module.css [hunk 3]
```diff
@@ -35,6 +35,8 @@
     margin: 0;
 
     color: var(--check-mark);
+
+    vertical-align: -0.125em;
```
Comment (awinogradov) [inline src/harmony/Checkbox/Checkbox.module.css:39]: magic number

File: src/harmony/Checkbox/Checkbox.tsx [hunk 4]
```diff
@@ -12,17 +13,16 @@ interface CheckboxProps extends React.InputHTMLAttributes<HTMLInputElement> {
 
 export const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(
     ({ label, className, view = 'default', ...rest }, ref) => (
-        <label
+        <Badge
```
Comment (awinogradov) [inline src/harmony/Checkbox/Checkbox.tsx:16]: Why we need to use Badge everywhere?

File: src/harmony/Dot/Dot.module.css [hunk 5]
```diff
@@ -2,11 +2,13 @@
     display: inline-block;
     border-radius: 50%;
     background-color: var(--dot-background);
+    vertical-align: middle;
 }
 
 .DotSmall {
     width: 0.375rem;
     height: 0.375rem;
+    vertical-align: 0.125em;
```
Comment (awinogradov) [inline src/harmony/Dot/Dot.module.css:11]: magic number

File: src/harmony/Radio/Radio.module.css [hunk 6]
```diff
@@ -31,9 +29,11 @@
 
     outline: none;
     padding: 0;
-    margin: 0.125rem; /* with this prop getting same size as Checkbox */
+    margin: 0.03125em; /* with this prop getting same size as Checkbox */
```
Comment (awinogradov) [inline src/harmony/Radio/Radio.module.css:32]: magic

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/harmony/Dot/Dot.module.css (hunk 5 above)
Comment (LamaEats) [reply src/harmony/Dot/Dot.module.css:11]: I can't explain it, however it correctly works
I saw this magic in [`font-awesome` docs](https://docs.fontawesome.com/web/add-icons/svg-symbols#make-sure-to-add-some-css) and tried to use it

File: src/harmony/Radio/Radio.module.css (hunk 6 above)
Comment (LamaEats) [reply src/harmony/Radio/Radio.module.css:32]: `<input type="radio" />`  has `15px` dimensions by sides
`<input type="checkbox" />`  has `16px` dimensions by sides

This magic number needs for correctly positioning inputs on the middle of the first line

File: src/harmony/Checkbox/Checkbox.module.css (hunk 3 above)
Comment (LamaEats) [reply src/harmony/Checkbox/Checkbox.module.css:39]: I can't explain it, however it correctly works
I saw this magic in [font-awesome docs](https://docs.fontawesome.com/web/add-icons/svg-symbols#make-sure-to-add-some-css) and tried to use it

File: src/harmony/Checkbox/Checkbox.tsx (hunk 4 above)
Comment (LamaEats) [reply src/harmony/Checkbox/Checkbox.tsx:16]: so that Badge, Checkbox, Radio looks the same

File: src/harmony/Badge/Badge.module.css (hunk 1 above)
Comment (LamaEats) [reply src/harmony/Badge/Badge.module.css:15]: 'cause `<Dot />` haven't `BaseIcon` wrapper

without these styles Badge with dot or icon will be looks are different
image

File: src/harmony/Checkbox/Checkbox.module.css (hunk 2 above)
Comment (LamaEats) [reply src/harmony/Checkbox/Checkbox.module.css:6]: inherit from? however Checkbox haven't sizes props

**Review by awinogradov — APPROVED** [review]

File: src/harmony/Radio/Radio.tsx [hunk 7]
```diff
@@ -2,23 +2,41 @@ import React, { forwardRef } from 'react';
...
-        })}
-    >
-        <input className={cn(classes.RadioInput)} type="radio" ref={ref} {...rest} defaultChecked={checked} />
-        {nullable(label, (labelComponent) => (
-            <span className={cn(classes.RadioLabel)}>{labelComponent}</span>
-        ))}
-    </label>
-));
+export const Radio = forwardRef<HTMLInputElement, RadioProps>(
+    ({ label, className, checked, strike, iconRight, lines, wordBreak, wordWrap, ellipsis, ...rest }, ref) => (
```
Comment (asabotovich) [inline src/harmony/Radio/Radio.tsx:19]: iconRight is quite unusual prop for Radio component..

File: src/harmony/Badge/Badge.tsx [hunk 8]
```diff
@@ -29,32 +41,48 @@ const sizeMap = {
...
-                ))}
-            </Text>
-        );
-    },
-);
+// eslint-disable-next-line prefer-arrow-callback
+export const Badge = forwardRef(function Badge<T extends keyof AllowedHTMLElements>(
+    props: BadgeProps & { as?: T } & AllowedHTMLElements[T],
+    ref: React.ForwardedRef<any>,
+) {
```
Comment (asabotovich) [inline src/harmony/Badge/Badge.tsx:48]: warning

**Review by asabotovich — APPROVED** [review]

**Pull request merged**